### PR TITLE
Implemented JWT Token

### PR DESCRIPTION
**Pull request by PiotrMichalak1**

### CODE REVIEW

File: src/main/java/org/example/auth/JwtAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import io.jsonwebtoken.Jwts;
+import org.example.models.JwtToken;
+import org.example.models.UserRole;
+
+import java.security.Key;
+import java.util.Optional;
+
+public class JwtAuthenticator implements Authenticator<String, JwtToken> {
+
+    Key key;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/auth/JwtAuthenticator.java:14]: private Key key; ?

File: src/main/java/org/example/controllers/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @POST
+    @Path("/login")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response login(LoginRequest loginRequest) {
+
+        try {
+            return Response.ok().entity(authService.login(loginRequest))
+                    .build();
+        } catch (SQLException e) {
+
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/AuthController.java:50]: please remove this line

File: src/main/java/org/example/controllers/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                                key = HttpHeaders.AUTHORIZATION,
+                                name = HttpHeaders.AUTHORIZATION,
+                                in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER
+                        )
+                }
+        )
+)
+public class AuthController {
+
+    AuthService authService;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/AuthController.java:35]: private?

File: src/main/java/org/example/models/JwtToken.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package org.example.models;
+
+import javax.security.auth.Subject;
+import java.security.Principal;
+
+public class JwtToken implements Principal {
+
+    UserRole userRole;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/JwtToken.java:8]: you even have getter ;) make it private

File: src/main/java/org/example/models/User.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package org.example.models;
+
+public class User {
+    private String username;
+    private String password;
+    int roleId;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/User.java:6]: why roleId is not private?

File: src/main/java/org/example/models/UserRole.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package org.example.models;
+
+import java.util.Map;
+
+public class UserRole {
+
+    public static final String HR = "HR";
+    public static final String MANAGER = "Manager";
+    public static final String SALES = "Sales";
+    int roleId;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/UserRole.java:10]: same

**Review by Alicja-Kainos — APPROVED** [review]

**Review by pedrosmake — APPROVED** [review]

**Pull request merged**